torch/distributed/checkpoint/default_planner.py

```diff
@@ -64,13 +64,19 @@ def __init__(
         self,
         flatten_state_dict: bool = True,
         flatten_sharded_tensors: bool = True,
-        dedup_replicated_tensors: bool = True,
+        dedup_replicated_tensors: Optional[bool] = None,
     ) -> None:
         self.flatten_state_dict = flatten_state_dict
         self.flatten_sharded_tensors = flatten_sharded_tensors
-        self.dedup_replicated_tensors = dedup_replicated_tensors
         self.mappings = {}
 
+        if dedup_replicated_tensors is not None:
```

Comment:
Setting `dedup_replicated_tensors` causes two separate assertions to fail during `create_default_global_save_plan` and `validate_global_save_plan`. Since the current state only supports a `true` value, I recommend we remove this option all together in a follow up PR.

On that note, I think we can now also remove `dedup_tensors` in favor of the new `dedup_save_plans`.
cc @fegin @wz337

Reply:
Ye. I think this is a good idea.

Reply:
Yup. I'm curious why not just update that API instead of creating a new one.

Reply:
This is more of a rename then a new API. I think renaming `dedup_tensors` -> `dedup_save_plans` makes sense since technically we are not only de-duplicating tensors, but any items in the SavePlan.

Just wanted to get some feedback on this PR before removing `dedup_tensors` in a follow up PR, since that will be a little more involved.